Update default campaign automatically (#5082)

gabina opened the pull request:
## What this PR does
This PR schedules an automatic job that runs at the beginning of the spring and fall in northern hemisphere (please double-check these dates):
- March, 21st
- September, 21st

The job performs a default campaign update using `DefaultCampaignUpdate` class. First of all, it determines the current term (for example, 'spring_2024' or 'fall_2024'). If the current term already exists as a campaign (this is manually created each year according to the issue description), then the current term is set as the default campaign.
Otherwise, default campaign is not updated. 

Notice that the update is only done when `wiki_education` feature is set to true, as I understand the scope of the issue is for Wiki Education Dashboard.

Closes #5082

## Open questions and concerns
- Double-check the spring and fall start dates.
- Determine if there is any fallback that we want to use when the current term doesn't exist as a campaign.

gabina commented:
Failing specs are not related to these changes.

ragesoss commented:
Code and tests look great! I'll give it a spin soon.

gabina commented on spec/workers/sidekiq_cron_jobs_spec.rb:
```diff
@@ -33,7 +33,7 @@
     before { allow(Features).to receive(:wiki_ed?).and_return(false) }
 
-    it 'run default campaign update' do
+    it 'do not run default campaign update' do
```
I just fixed this, since the description was incorrectly saying "run default campaign update", when what I wanted to specified was the opposite.

ragesoss commented on lib/deafult_campaign_update.rb:
```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
```
Filename is misspelled.

gabina replied:
Fixed in fd9a70f9c5cded399d1a84ae6c04271312717bd8

ragesoss commented:
We don't actually want to switch these based on the current season, but rather based on the academic calendar where 'spring' is the semester that starts in January. I think the right dates for this feature are January 1 through June 30 for spring, and July 1 through December 31 for fall.

gabina commented:
> We don't actually want to switch these based on the current season, but rather based on the academic calendar where 'spring' is the semester that starts in January. I think the right dates for this feature are January 1 through June 30 for spring, and July 1 through December 31 for fall.

oh that makes sense! I changed the dates on commit 927f8072c3583407f22aab62746d0339f192b4e7, and added some comments to clarify that dates are related to spring/fall semesters based on academic calendar.

gabina commented:
I closed this PR by mistake and I don't think I'm able to reopen it now. But it should be re-opened, as it's ready for review and merge.
